src/pages/feed/components/FilterBottomSheet/FilterBottomSheet.tsx

```diff
@@ -0,0 +1,266 @@
+  skillCategoryResponses: Idea['skillCategoryResponses'];
+};
+
+const FilterBottomSheet = ({
+  open,
+  onClose,
+  onApply,
+  branches,
+  purposes,
+  recruitmentPlaces,
+  skillCategoryResponses,
+}: Props) => {
+  const { filterParams, updateFilterParams, resetFilterParams } = useFilterParams();
+
+  const branchOptions = branches.map((properties) => ({
+    checked: filterParams?.branchIds?.includes(properties.id) ? true : false,
+    ...properties,
+  }));
+  const purposeOptions = purposes.map((properties) => ({
+    checked: filterParams?.purposeIds?.includes(properties.id) ? true : false,
+    ...properties,
+  }));
+  const { checkboxValue, onChangeCheckbox } = useCheckbox({
+    branches: branchOptions,
+    purposes: purposeOptions,
+  });
+
+  const cooperationWayOptions =
+    filterParams?.cooperationWay === undefined
+      ? cooperationWays.map((properties) => {
+          // 필터 선택 안 되어 있을 경우 상관없음이 기본값(id === 1)
+          return properties.id === 1 ? { checked: true, ...properties } : { checked: false, ...properties };
+        })
+      : cooperationWays.map((properties) => ({
+          checked: filterParams?.cooperationWay === properties.name ? true : false,
+          ...properties,
+        }));
+  const { radioValue, onChangeRadio } = useRadio({
+    cooperationWays: cooperationWayOptions,
+  });
+
+  const getSkillCategory1DepthFrom2DepthSkillId = (id: number) => {
+    const skillCategory1Depth = skillCategoryResponses.find((item) =>
+      item.skillResponses.find((skill) => skill.id === id),
+    );
+
+    if (skillCategory1Depth === undefined) {
+      throw new Error('skillCategory1Depth skill category not found');
+    }
+    return skillCategory1Depth;
+  };
+
+  const get2DepthNameFrom2DepthId = (id: number) => {
+    const name = skillCategoryResponses
+      .find((item) => item.skillResponses.find((skill) => skill.id === id))
+      ?.skillResponses.find((skill) => skill.id === id)?.name;
+
+    if (name === undefined) {
+      throw new Error('2depth skill category not found');
+    }
+
+    return name;
+  };
+
+  const { dropdownValue, onClickDropdown } = useDropdown({
+    recruitmentPlace: recruitmentPlaces.find((place) => place.id === filterParams?.recruitmentPlaceId)?.name ?? '',
+  });
+
+  const { dropdownValue: skillCategory1DepthDropDownValue, onClickDropdown: onCklickSkillCategory1DepthDropDown } =
+    useDropdown({
+      skillCategory1Depth:
+        filterParams?.skillCategoryIds?.[0] !== undefined
+          ? getSkillCategory1DepthFrom2DepthSkillId(filterParams?.skillCategoryIds?.[0]).name
+          : undefined ?? '',
+    });
+
+  const { dropdownValue: skillCategory2DepthDropDownValue, onClickDropdown: onCklickSkillCategory2DepthDropDown } =
+    useDropdown({
+      skillCategory2Depth:
+        filterParams?.skillCategoryIds?.[0] !== undefined
+          ? get2DepthNameFrom2DepthId(filterParams?.skillCategoryIds?.[0])
+          : undefined ?? '',
+    });
```

Comment:
한 폼당 하나의 훅으로 상태를 관리하고자 했던게 제 use 훅 시리즈여서 위 로직은 아래처럼 수정해볼 수 있겠습니다만, 적용하지는 말아주세요~

```suggestion
  const { dropdownValue, onClickDropdown } = useDropdown({
    recruitmentPlace: recruitmentPlaces.find((place) => place.id === filterParams?.recruitmentPlaceId)?.name ?? '',
    skillCategory1Depth : filterParams?.skillCategoryIds?.[0] !== undefined
          ? getSkillCategory1DepthFrom2DepthSkillId(filterParams?.skillCategoryIds?.[0]).name
          : undefined ?? '',
    skillCategory2Depth : filterParams?.skillCategoryIds?.[0] !== undefined
          ? get2DepthNameFrom2DepthId(filterParams?.skillCategoryIds?.[0])
          : undefined ?? '',
  });
```
필터링이 되어있으면 이전 필터링 값을 적용해야하다보니 불필요한 로직이 많이 발생되는군요. 문제점을 잘 파악했습니다.